cql3/restrictions/statement_restrictions.cc

```diff
@@ -683,6 +692,54 @@ std::vector<const column_definition*> statement_restrictions::get_column_defs_fo
     return column_defs_for_filtering;
 }
 
+void statement_restrictions::add_restriction(const expr::binary_operator& restr, schema_ptr schema, bool allow_filtering, bool for_view) {
+    if (restr.op == expr::oper_t::IS_NOT) {
+        // Handle IS NOT NULL restrictions seperately
+        add_is_not_restriction(restr, schema, for_view);
+    } else if (expr::is_multi_column(restr)) {
+        // Multi column restrictions are only allowed on clustering columns
+        add_multi_column_clustering_key_restriction(restr);
+    } else if (has_token(restr)) {
+        // Token always restricts the partition key
+        add_token_partition_key_restriction(restr);
+    } else if (expr::is_single_column_restriction(restr)) {
+        const column_definition* def = get_the_only_column(restr).col;
+        if (def->is_partition_key()) {
+            add_single_column_parition_key_restriction(restr, schema, allow_filtering, for_view);
+        } else if (def->is_clustering_key()) {
+            add_single_column_clustering_key_restriction(restr, schema, allow_filtering);
+        } else {
+            add_single_column_nonprimary_key_restriction(restr);
+        }
+    } else {
+        throw exceptions::invalid_request_exception(format("Unhandled restriction: {}", restr));
+    }
+}
+
+void statement_restrictions::add_is_not_restriction(const expr::binary_operator& restr, schema_ptr schema, bool for_view) {
+    // TODO(restrictions): Implement this
```

Comment:
Maybe it's better to go with `on_internal_error` instead of silently letting this call slip? That applies to all unimplemented stubs

Reply:
I guess the implementations will be folded in later.

Reply:
Doing `on_internal_error` would break bisectability because the functions are called despite being unimplemented.

Folded all `add_restriction` stuff into a single commit as requested by Avi